### ISSUE TITLE
[gatsby-remark-images-contentful] No stack trace on unsupported image format

### ISSUE THREAD

**Issue by bartveneman** [gatsbyjs/gatsby#32511]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

We've spent a couple of hours debugging our Gatsby builds because of this error: `Error: Input buffer contains unsupported image format`. We had just added `gatsby-remark-images-contentful` and then our builds started failing. It took a couple of hours to figure out that someone has accidentally prefixed a markdown link with a `!`, turning it into an image instead. The url linked to a `.zip` file: `![alt here](https://images.ctfassets.net/zar1ypr5qpcx/3OxjYd3EKsLDfXiPOLc6k0/aafe04e868ac34660c0a9d04dfb7e9d3/kerstkaart-lamp.zip)` and caused the complete build to crash with no traces where to start looking for errors.
Impact: several people, including those from Gatsby spending hours to figure out what might be going on. A clear stack trace would have sped up the process quite a lot.

### Reproduction Link

https://github.com/bartveneman/gatsby-repro-contentful-images-failure

### Steps to Reproduce

1. Add an image with a link to an unsupported image type to a markdown input (locally or in Contentful), for example: `![alt here](https://images.ctfassets.net/zar1ypr5qpcx/3OxjYd3EKsLDfXiPOLc6k0/aafe04e868ac34660c0a9d04dfb7e9d3/kerstkaart-lamp.zip)`
2. `npm run dev` or `npm run build`
3. The error is right there

### Expected Result

A clear error message telling *which* image type isn't supported, which url it was trying to optimize and maybe even the location of the Markdown the image was in.

### Actual Result

Only a minial error message is shown, without any leads where to investigate further: `Error: Input buffer contains unsupported image format`

### Environment

```shell
System:
    OS: macOS 11.4
    CPU: (8) x64 Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.15.3 - ~/.nvm/versions/node/v14.15.3/bin/node
    Yarn: 1.22.5 - ~/.yarn/bin/yarn
    npm: 7.5.2 - ~/.nvm/versions/node/v14.15.3/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 88.0.4324.150
    Firefox: 89.0.2
    Safari: 14.1.1
  npmPackages:
    gatsby: ^3.10.1 => 3.10.1 
    gatsby-plugin-sharp: ^3.10.2 => 3.10.2 
    gatsby-remark-images-contentful: ^4.7.0 => 4.7.0 
    gatsby-source-filesystem: ^3.10.0 => 3.10.0 
    gatsby-transformer-remark: ^4.7.0 => 4.7.0
```

### Config Flags

N/A

**Comment by LekoArts** [comment]
Hi, thanks for the issue!

The plugin fails here: https://github.com/gatsbyjs/gatsby/blob/7d3b47692a56c387edcaf6a841d9172225a1bfe9/packages/gatsby-remark-images-contentful/src/index.js#L84

So you could wrap this with a `try/catch` like so:

```js
let metadata
try {
  metadata = await metaReader.metadata();
} catch (error) {
  reporter.panic(`The image "${node.url}" (with alt text: "${node.alt}") doesn't appear to be a supported image format.`, error)
}
```

We'd be happy to review a PR adding this + a test ensuring this functionality.

**Comment by axe312ger** [comment]
I'll have a look :eyes:

**Comment by axe312ger** [comment]
Should be fixed in #32530. Please have a look :)

**Comment by bartveneman** [comment]
Fix works like a charm!
